### PR TITLE
Some QuantStack libraries

### PR DESCRIPTION
**Pull request by mredaelli**
###### Motivation for this change
Missing libraries from the QuantStack ecosystem

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/gast/default.nix
Comment (FRidh) [inline pkgs/development/python-modules/gast/default.nix]: these two should go into `buildPythonPackage`

File: pkgs/development/python-modules/gast/default.nix
Comment (FRidh) [inline pkgs/development/python-modules/gast/default.nix]: and this removed

File: pkgs/development/libraries/xsimd/default.nix
Comment (FRidh) [inline pkgs/development/libraries/xsimd/default.nix]: version is not an attribute of `stdenv.mkDerivation`

File: pkgs/development/libraries/xsimd/default.nix
Comment (FRidh) [inline pkgs/development/libraries/xsimd/default.nix]: maybe use `fetchFromGitHub`? At least insert `${version}`

File: pkgs/development/libraries/xtensor/default.nix
Comment (FRidh) [inline pkgs/development/libraries/xtensor/default.nix]: same

File: pkgs/development/libraries/xtl/default.nix
Comment (FRidh) [inline pkgs/development/libraries/xtl/default.nix]: same

File: pkgs/development/python-modules/gast/default.nix
Comment (FRidh) [inline pkgs/development/python-modules/gast/default.nix]: fetching from master is a bad idea, not reproducible

File: pkgs/development/python-modules/pythran/default.nix
Comment (FRidh) [inline pkgs/development/python-modules/pythran/default.nix]: into `buildPythonPackage`

File: pkgs/development/python-modules/pythran/default.nix
Comment (FRidh) [inline pkgs/development/python-modules/pythran/default.nix]: and remove this

File: pkgs/top-level/python-packages.nix
Comment (FRidh) [inline pkgs/top-level/python-packages.nix]: please don't include any whitespace changes

**Review by FRidh — CHANGES_REQUESTED** [review]

**Comment by FRidh** [comment]
I have to think about what we should do with networkx considering we typically only permit a single version of a package.

**Comment by mredaelli** [comment]
Thanks for detailed review. Much needed: it's my first try :)

File: pkgs/development/python-modules/gast/default.nix
Comment (mredaelli) [reply pkgs/development/python-modules/gast/default.nix]: Yeah, but it doesn't have any release or tag. Should I go for a commit?

Comment (FRidh) [reply pkgs/development/python-modules/gast/default.nix]: Yep. Do try to find the commit corresponding to the version.

**Comment by mredaelli** [comment]
Should be ok now, except for networkx. Let me know if I should change something

**Comment by dotlambda** [comment]
Networkx 2 is already in the staging branch. We should wait until that is merged.

**Comment by dotlambda** [comment]
Networkx v2 is in master. Please rebase and remove your expression for v2.

**Comment by mmahut** [comment]
What is the status of this pull request?

**Comment by FRidh** [comment]
Nothing happening so closing.

**Pull request closed without merging**
